PopularMoviesApp/MoviesDemo/MainViewController.swift

```diff
@@ -0,0 +1,65 @@
+//  MainViewController.swift
+//  Movies Demo
+//
+//  Created by Ahmed Hamdy Gomaa on 4/15/18.
+//  Copyright © 2018 Ahmed Hamdy Gomaa. All rights reserved.
+//
+
+import UIKit
+import Kingfisher
+
+class MainViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+    @IBOutlet weak var mainTableView: UITableView!
+    var movieViewModel = MainViewModel()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        movieViewModel.clearImageCache()
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 1
+    }
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return movieViewModel.apiKeyWords.count
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let movieCell = tableView.dequeueReusableCell(withIdentifier: "MainCell", for: indexPath) as! MainTableViewCell
+        configureCell(movieCell, at: indexPath)
+        return movieCell
+    }
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return self.view.frame.height/2
+    }
+
+    private func configureCell(_ cell: MainTableViewCell, at indexPath: IndexPath) {
+        let categories = ["Popular", "Top Rated", "Favourites"]
+        func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+            let movieCell = tableView.dequeueReusableCell(withIdentifier: "MainCell", for: indexPath) as! MainTableViewCell
+            configureCell(movieCell, at: indexPath)
+            return movieCell
+        }
+        
+        private func configureCell(_ cell: MainTableViewCell, at indexPath: IndexPath) {
+            let categories = ["Popular", "Top Rated", "Favourites"]
```

Comment:
remove this line from the method as no need to recreate it every time method called

Reply:
no updates yet